extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -1382,8 +1386,14 @@ export class AmpA4A extends AMP.BaseElement {
    * variables of the form { name: val }.  It is up to the subclass what to
    * do with those variables.
    *
-   * @param {string} unusedEventName
+   * TODO(tdrl): This method and its instantiations should go away once
+   * we've verified that analytics is working correctly for performance
+   * monitoring and we can deprecate the direct CSI ping mechanism.
+   *
+   * @param {string} eventName
    * @param {!Object<string, string|number>=} opt_extraVariables
    */
-  emitLifecycleEvent(unusedEventName, opt_extraVariables) {}
+  emitLifecycleEvent(eventName, opt_extraVariables) {
+    triggerAnalyticsEvent(this.ampDoc_, eventName, opt_extraVariables);
```

Comment:
Do we have a function to generate `eventName` here? Otherwise a general eventName doesn't work here. 
Say we have  two ads, A and B. Both of them send event `adRequestStart` at some time. The calling of `#triggerAnalyticsEvent` from A will also trigger the handler of `B` because it will trigger all events handler under the same eventName.